pallets/nfts/src/lib.rs

```diff
@@ -1295,80 +1340,128 @@ pub mod pallet {
 		///
 		/// Weight: `O(1)`
 		#[pallet::call_index(15)]
-		#[pallet::weight(T::WeightInfo::approve_transfer())]
+		#[pallet::weight(T::WeightInfo::approve_transfer(maybe_item.is_some() as u32))]
 		pub fn approve_transfer(
 			origin: OriginFor<T>,
 			collection: T::CollectionId,
-			item: T::ItemId,
+			maybe_item: Option<T::ItemId>,
 			delegate: AccountIdLookupOf<T>,
 			maybe_deadline: Option<BlockNumberFor<T>>,
 		) -> DispatchResult {
-			let maybe_check_origin = T::ForceOrigin::try_origin(origin)
-				.map(|_| None)
-				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
 			let delegate = T::Lookup::lookup(delegate)?;
-			Self::do_approve_transfer(
-				maybe_check_origin,
-				collection,
-				item,
-				delegate,
-				maybe_deadline,
-			)
+			match maybe_item {
+				Some(item) => {
+					let maybe_check_origin =
+						T::ForceOrigin::try_origin(origin).map(|_| None).or_else(|origin| {
+							ensure_signed(origin).map(Some).map_err(DispatchError::from)
+						})?;
+					Self::do_approve_transfer(
+						maybe_check_origin,
+						collection,
+						item,
+						delegate,
+						maybe_deadline,
+					)
+				},
+				None => {
+					let origin = ensure_signed(origin)?;
```

Comment:
Why is there no consideration for ForceOrigin here as above?

Reply:
Explained somewhere

Reply:
https://github.com/r0gue-io/pop-node/pull/387#discussion_r1865631705

Reply:
First, approving and cancelling an approval are different from each other in respect with this discussion. 

As for whether it makes sense for approving. The owner's account has to be known in order to take the deposit from the right address. That doesn't make it possible to allow the ForceOrigin.